src/calendar/index.js

```diff
@@ -88,6 +88,12 @@ class Calendar extends Component {
 
     this.shouldComponentUpdate = shouldComponentUpdate;
   }
+
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
@ questions: 
1) Why not use `shouldComponentUpdate` as with all other props instead of adding the 'componentDidUpdate' method just for it?
2) When you what to update the calendar with a new 'current', did you try to pass the 'key' prop with this dynamic data?

Reply:
Hm, I'm not sure I follow the idea of a `key` prop, I don't see that prop listed in the prop type definitions.

My understanding of `shouldComponentUpdate` is that it controls whether the component should rerender or not, but re-rendering in this case would not have an effect on `currentMonth` in state, because that bit of state is only set in the constructor (before the component mounts) or when responding to a touch event

Reply:
`key` is a React component prop that should trigger rerender when changed.